stdlib/public/core/Dictionary.swift

```diff
@@ -890,13 +892,32 @@ extension Dictionary {
       _variantBuffer: _variantBuffer.mapValues(transform))
   }
 
-  /// Returns a new dictionary containing the keys of this dictionary with the
-  /// values transformed by the given closure.
+  /// Returns a new dictionary containing only the key-value pairs that have
+  /// non-`nil` values as the result from the transform by the given closure.
+  ///
+  /// Use this method to receive a dictionary of nonoptional values when your
```

Comment:
@natecook1000 should it be "nonoptional" or "non-optional"? I would vote for the latter, since it's a little easier to parse. Just wondering, no big deal.

Reply:
"nonoptional" is the one we use elsewhere, currently — updating the other uses over in #18263. Thanks!